libsolidity/interface/StandardCompiler.cpp

```diff
@@ -557,32 +553,32 @@ std::optional<Json::Value> checkOutputSelection(Json::Value const& _outputSelect
 
 /// Validates the optimizer settings and returns them in a parsed object.
 /// On error returns the json-formatted error message.
-std::variant<OptimiserSettings, Json::Value> parseOptimizerSettings(Json::Value const& _jsonInput)
+std::variant<OptimiserSettings, Json> parseOptimizerSettings(Json const& _jsonInput)
 {
 	if (auto result = checkOptimizerKeys(_jsonInput))
 		return *result;
 
 	OptimiserSettings settings = OptimiserSettings::minimal();
 
-	if (_jsonInput.isMember("enabled"))
+	if (_jsonInput.contains("enabled"))
 	{
-		if (!_jsonInput["enabled"].isBool())
+		if (!_jsonInput["enabled"].is_boolean())
 			return formatFatalError("JSONError", "The \"enabled\" setting must be a Boolean.");
 
-		if (_jsonInput["enabled"].asBool())
+		if (_jsonInput["enabled"].get<bool>())
 			settings = OptimiserSettings::standard();
 	}
 
-	if (_jsonInput.isMember("runs"))
+	if (_jsonInput.contains("runs"))
 	{
-		if (!_jsonInput["runs"].isUInt())
+		if (!_jsonInput["runs"].is_number_unsigned())
 			return formatFatalError("JSONError", "The \"runs\" setting must be an unsigned number.");
-		settings.expectedExecutionsPerDeployment = _jsonInput["runs"].asUInt();
+		settings.expectedExecutionsPerDeployment = _jsonInput["runs"].get<Json::number_unsigned_t>();
```

Comment:
We should validate the range of the value or have a static assert that types are the same.